config/ompi_setup_java.m4

```diff
@@ -74,10 +74,22 @@ AC_DEFUN([OMPI_SETUP_JAVA_BINDINGS],[
         # header file needs this file, so we need to check for
         # it/include it in our sources when compiling on Mac).
         AC_CHECK_HEADERS([TargetConditionals.h])
+
+        # dladdr is required to build the full path to libmpi on OS X 10.11 aka El Capitan
+        # OPAL_CHECK_PACKAGE([mpi_java], [dlfcn.h], [dl], [dladdr], [], [], [], [OMPI_HAVE_DLADDR=1], [OMPI_HAVE_DLADDR=0])
+        #AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <dlfcn.h> ]],
+                                        #[[Dl_info info; dladdr((void *)0, &info);]])],
+                       #[OMPI_HAVE_DLADDR=1],
+                       #[OMPI_HAVE_DLADDR=0])
+        AC_CHECK_TYPE([Dl_info], [OMPI_HAVE_DLADDR=1], [OMPI_HAVE_DLADDR=0], [[#include <dlfcn.h>]])
```

Comment:
Can you replace most of the changes in this file with `AC_CHECK_TYPES`?  We do this kind of thing in a bunch of places (even though it will use the GNU convention of `#define` vs. `#undef`).